meson.build

```diff
@@ -1470,9 +1470,7 @@ elif compression == 'lz4' and not have_lz4
 elif compression == 'xz' and not have_xz
         error('default-compression=xz requires xz')
 endif
-conf.set10('DEFAULT_COMPRESSION_ZSTD', compression == 'zstd')
-conf.set10('DEFAULT_COMPRESSION_LZ4', compression == 'lz4')
-conf.set10('DEFAULT_COMPRESSION_XZ', compression == 'xz')
+conf.set('DEFAULT_COMPRESSION', 'OBJECT_COMPRESSED_@0@'.format(compression.to_upper()))
```

Comment:
compat.h is no longer self-defined now (it was raised as a point in the previous PR)

Reply:
I think you mean `config.h` and `config.h` already contains defines for `ANSI_OK_COLOR` and `STATUS_UNIT_FORMAT_DEFAULT` to enum identifiers or other preprocessor macros not defined in `config.h`.

Reply:
It's "self-defined" in the sense that you can include it and it is valid. Some of the defines are not meaningful without other headers, but that's OK.

Reply:
won't the preprocessor barf because OBJECT_COMPRESSED_XYZ does not exist?